tests/unittests/config/test_apt_source_v1.py

```diff
@@ -44,51 +46,46 @@ def update_package_sources(self):
         return
 
 
-class TestAptSourceConfig(TestCase):
+class TestAptSourceConfig:
     """TestAptSourceConfig
     Main Class to test apt_source configs
     """
 
     release = "fantastic"
-
-    def setUp(self):
-        super(TestAptSourceConfig, self).setUp()
-        self.tmp = tempfile.mkdtemp()
-        self.addCleanup(shutil.rmtree, self.tmp)
-        self.aptlistfile = os.path.join(self.tmp, "single-deb.list")
-        self.aptlistfile2 = os.path.join(self.tmp, "single-deb2.list")
-        self.aptlistfile3 = os.path.join(self.tmp, "single-deb3.list")
-        self.join = os.path.join
-        self.matcher = re.compile(ADD_APT_REPO_MATCH).search
-        # mock fallback filename into writable tmp dir
-        self.fallbackfn = os.path.join(
-            self.tmp, "etc/apt/sources.list.d/", "cloud_config_sources.list"
+    matcher = re.compile(ADD_APT_REPO_MATCH).search
+
+    @pytest.fixture
+    def apt_lists(self, tmpdir):
+        p1 = os.path.join(tmpdir, "single-deb.list")
+        p2 = os.path.join(tmpdir, "single-deb2.list")
+        p3 = os.path.join(tmpdir, "single-deb3.list")
+        return p1, p2, p3
+
+    @pytest.fixture
+    def fallback_path(self, tmpdir):
+        return os.path.join(
+            tmpdir, "etc/apt/sources.list.d/", "cloud_config_sources.list"
         )
 
-        self.cloud = get_cloud()
-
-        rpatcher = mock.patch("cloudinit.util.lsb_release")
-        get_rel = rpatcher.start()
-        get_rel.return_value = {"codename": self.release}
-        self.addCleanup(rpatcher.stop)
-        apatcher = mock.patch("cloudinit.util.get_dpkg_architecture")
-        get_arch = apatcher.start()
-        get_arch.return_value = "amd64"
-        self.addCleanup(apatcher.stop)
-        subp_patcher = mock.patch.object(
-            subp, "subp", return_value=("PPID   PID", "")
+    @pytest.fixture(autouse=True)
+    def common_mocks(self, mocker):
```

Comment:
better naming that the 'setup' I had in my tests. I'll likely rename setup->common_mocks in the deb822 support PR that will be up shortly.